e2e/nomostest/nt.go

```diff
@@ -568,123 +564,83 @@ func (nt *NT) MustDeleteGatekeeperTestData(file, name string) {
 	nt.MustKubectl("delete", "-f", absPath, "--ignore-not-found", "--wait")
 }
 
-// PortForwardOtelCollector forwards the otel-collector pod.
-func (nt *NT) PortForwardOtelCollector() {
-	// Retry otel-collector port-forwarding in case it is in the process of upgrade.
-	took, err := retry.Retry(nt.DefaultWaitTimeout, func() error {
-		pod, err := nt.KubeClient.GetDeploymentPod(ocmetrics.OtelCollectorName, ocmetrics.MonitoringNamespace, nt.DefaultWaitTimeout)
-		if err != nil {
-			return err
-		}
-		// The otel-collector forwarding port needs to be updated after otel-collector restarts or starts for the first time.
-		// It sets otelCollectorPodName and otelCollectorPort to point to the current running pod that forwards the port.
-		if pod.Name != nt.otelCollectorPodName {
-			ctx, cancel := context.WithCancel(nt.Context)
-			port, err := nt.ForwardToFreePort(ctx, ocmetrics.MonitoringNamespace, pod.Name,
-				testmetrics.OtelCollectorMetricsPort)
-			if err != nil {
-				cancel()
-				return err
-			}
-			nt.T.Cleanup(cancel) // stop port-forward
-			nt.otelCollectorPort = port
-			nt.otelCollectorPodName = pod.Name
-		}
-		return nil
-	})
-	if err != nil {
-		nt.T.Fatal(err)
-	}
-	nt.T.Logf("took %v to wait for otel-collector port-forward", took)
+func (nt *NT) newPortForward(ns, deployment, port string, opts ...portforwarder.PortForwardOpt) *portforwarder.PortForwarder {
+	ctx, cancel := context.WithCancel(nt.Context)
+	nt.T.Cleanup(cancel)
+	return portforwarder.NewPortForward(
+		ctx,
+		nt.KubeClient,
+		nt.Watcher,
+		nt.Logger,
+		nt.DefaultWaitTimeout,
+		nt.kubeconfigPath,
+		ns, deployment, port,
+		opts...,
+	)
 }
 
-// PortForwardGitServer forwards the git-server deployment to a port.
-func (nt *NT) PortForwardGitServer() {
-	nt.T.Helper()
-
-	pod, err := nt.KubeClient.GetDeploymentPod(testGitServer, testGitNamespace, nt.DefaultWaitTimeout)
-	if err != nil {
-		nt.T.Fatal(err)
-	}
-	podName := pod.Name
-
-	if podName != nt.gitRepoPodName {
-		ctx, cancel := context.WithCancel(nt.Context)
-		port, err := nt.ForwardToFreePort(ctx, testGitNamespace, podName, 22)
-		if err != nil {
-			cancel()
-			nt.T.Fatal(err)
-		}
-		nt.T.Cleanup(cancel) // stop port-forward
-		nt.gitRepoPort = port
-		nt.gitRepoPodName = podName
+// portForwardOtelCollector forwards the otel-collector deployment to a port.
+func (nt *NT) portForwardOtelCollector() {
+	if nt.otelCollectorPortForward != nil {
+		nt.T.Fatal("otel collector port forward already initialized")
 	}
+	nt.otelCollectorPortForward = nt.newPortForward(
```

Comment:
What if we just construct these in the test setup func? Are they called from elsewhere? Does it need to be its own function?

Reply:
nvm. i see portForwardGitServer is more complex and it's nice to be consistent.